### PR TITLE
Login api integration

### PR DESCRIPTION
**Pull request by sansanthi**
**Trello Ticket(s)**

- https://trello.com/c/UC79hrQq/22-frontend-implement-api-call-for-the-user-login

**Related PR(s)**

- #5 

**Description**

- integrate login api  with redux-toolkit 
- create routes for routing

**Screenshots**

https://user-images.githubusercontent.com/28820517/232206410-8b612831-3886-4833-97f3-36e9c16afc11.mov

### CODE REVIEW

File: src/features/login/ui/LoginForm.tsx [hunk 1]
```diff
@@ -1,14 +1,143 @@
...
+        console.log(response.data);
+        if (response.data.length === 0) {
+          alert("no user found");
+          setIsCorrect("false");
+        } else {
+          alert(response.data);
+          setIsCorrect("true");
+        }
+      });
+  };
```
Comment (AungKhantMin21) [inline src/features/login/ui/LoginForm.tsx]:
```suggestion
  const getUser = (email: string, password: string) => {
    axios
      .get<User[]>(
        `http://localhost:3001/Users?email=${email}&password=${password}`
      )
      .then((response) => {
        console.log(response.data);
        if (response.data.length === 0) {
          alert("no user found");
          setIsCorrect("false");
        } else {
          alert(response.data);
          setIsCorrect("true");
        }
      });
  };
```
This function is written for testing with JSON server and I think this is no longer needed.

File: src/features/login/ui/LoginForm.tsx [hunk 2]
```diff
@@ -1,14 +1,124 @@
...
+  const [isCorrect, setIsCorrect] = useState<string>("uncheck");
+
+  const { loading, token, error } = useAppSelector((state) => state.login);
+  const dispatch = useAppDispatch();
+
+  const naviagte = useNavigate();
+
+  useEffect(() => {
+    if (token) {
+      console.log("user token in login:", token);
```
Comment (AungAyeThan) [inline src/features/login/ui/LoginForm.tsx]: lets remove console log 😬

File: src/slices/loginSlice.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+const initialState: InitialState = {
+  loading: false,
+  userInfo: {},
+  token: "",
+  error: "",
+  success: "",
+};
+
+const LOGIN_URL = "http://localhost:5000/login";
```
Comment (AungAyeThan) [inline src/slices/loginSlice.ts]: better to use env variable with .env and reuse with something like {process.env. API_URL}

File: .env [hunk 4]
```diff
@@ -0,0 +1 @@
+REACT_APP_LOGIN_URL = http://localhost:5000/login
```
Comment (AungAyeThan) [inline .env:1]: better to put HOSTNAME only for env but lets change it in next PR 😬

**Review by AungAyeThan — APPROVED** [review]

**Review by Kaung-Htet-Naing — APPROVED** [review]

**Pull request merged**